File: MiniMemory/app/src/main/java/com/marinesnow/minimemory/storehelp.java
package com.marinesnow.minimemory;

/**
 * Created by mei on 2017/7/2.
 */

import android.content.SharedPreferences;
import android.content.Context;

public class storehelp
{
    public static final String MAIN_SET = "main";

    private SharedPreferences sp;
    private SharedPreferences.Editor se;

    public storehelp(Context cp,String name){
        sp = cp.getSharedPreferences(name,0);
        se = sp.edit();
    }

    public storehelp(Context c){
        sp = c.getSharedPreferences(MAIN_SET,0);
        se = sp.edit();
    }

    public boolean con(String key){
        return sp.contains(key);
    }

    public void put(String key, int value)
    {
        se.putInt(key,value);
        se.commit();
    }

    public int getInt(String key){
        return sp.getInt(key,0);
    }

    public void put(String key ,String value){
        se.putString(key ,value);
        se.commit();
    }
    public String getS(String key){
        return sp.getString(key,"null");
    }

    public void put(String key ,float value){
        se.putFloat(key,value);
        se.commit();
    }

    public float getF(String key){
        return sp.getFloat(key,1.0f);

    }
}

File: MiniMemory/app/src/main/java/com/marinesnow/minimemory/storehelp.kt
package com.marinesnow.minimemory

/**
 * Created by mei on 2017/7/2.
 */

import android.content.SharedPreferences
import android.content.Context

class storehelp {

    private var sp: SharedPreferences? = null
    private var se: SharedPreferences.Editor? = null

    constructor(cp: Context, name: String) {
        sp = cp.getSharedPreferences(name, 0)
        se = sp!!.edit()
    }

    constructor(c: Context) {
        sp = c.getSharedPreferences(MAIN_SET, 0)
        se = sp!!.edit()
    }

    fun con(key: String): Boolean {
        return sp!!.contains(key)
    }

    fun put(key: String, value: Int) {
        se!!.putInt(key, value)
        se!!.commit()
    }

    fun getInt(key: String): Int {
        return sp!!.getInt(key, 0)
    }

    fun put(key: String, value: String) {
        se!!.putString(key, value)
        se!!.commit()
    }

    fun getS(key: String): String {
        return sp!!.getString(key, "null")
    }

    fun put(key: String, value: Float) {
        se!!.putFloat(key, value)
        se!!.commit()
    }

    fun getF(key: String): Float {
        return sp!!.getFloat(key, 1.0f)

    }

    companion object {
        val MAIN_SET = "main"
    }
}
